### PR TITLE
Ensure generated code has the correct header to be picked up by code analysis.

### LINKED ISSUE #848: Generated file is still analyzed for Code Analysis
Every file in "Generated Files\*.cs" is a generated file, so they should not be analyzed for Code Analysis warnings. Using a projection with CsWinRTIncludes and CsWinRTWindowsMetadata should not add a bunch of warnings, as it does today.
**Describe the bug**

**To Reproduce**
1. Project a winmd using CsWinRTIncludes and CsWinRTWindowsMetadata.
2. There will be dozens of SA1400, SA1127, SA1308, etc...

**Expected behavior**
No warnings for generated code.

**Version Info**
CS WinRT 1.2.6

**Additional context**
Maybe leveraging [GeneratedCodeAttribute](https://docs.microsoft.com/en-us/dotnet/api/system.codedom.compiler.generatedcodeattribute?redirectedfrom=MSDN&view=net-5.0) is all that is needed?

### PR DESCRIPTION
**Pull request by jlaanstra**
Fixes #848

### CODE REVIEW

File: src/cswinrt/main.cpp [hunk 1]
```diff
@@ -321,7 +321,15 @@ Where <spec> is one or more of:
                 group.add([&componentActivatableClasses, &projectionFileWritten]
                 {
                     writer wm;
-                    wm.write("// This file was generated by cswinrt.exe version %", VERSION_STRING);
+                    wm.write(R"(//------------------------------------------------------------------------------
+// <auto-generated>
```
Comment (manodasanW) [inline src/cswinrt/main.cpp:325]: Does the auto-generated tag cause the code analysis tooling to not pick it up?  Is that documented somewhere?

Comment (jlaanstra) [reply src/cswinrt/main.cpp:325]: I can't immediately find the public docs for it, but the Roslyn code is here: https://github.com/dotnet/roslyn/blob/cf55f3a58e47298426fa971d3bd9d8857c746c65/src/Compilers/Core/Portable/InternalUtilities/GeneratedCodeUtilities.cs

File: src/cswinrt/main.cpp [hunk 2]
```diff
@@ -321,7 +321,15 @@ Where <spec> is one or more of:
...
+//     This file was generated by cswinrt.exe version %
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+)", VERSION_STRING);
```
Comment (manodasanW) [inline src/cswinrt/main.cpp:332]: Can you also add a similar write line below to `writer eventHelperWriter("WinRT");`.  It should probably have one, but seems to be have been missed before.

Comment (jlaanstra) [reply src/cswinrt/main.cpp:332]: Done. Let me know if this is what you meant.

Comment (manodasanW) [reply src/cswinrt/main.cpp:332]: Thanks, that is what I meant.

**Review by manodasanW — APPROVED** [review]

**Comment by manodasanW** [comment]
Thanks!

**Pull request merged**